kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -34,21 +41,25 @@
 import io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider.PortPerBrokerClusterNetworkAddressConfigProvider;
 import io.kroxylicious.proxy.service.HostPort;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
+
 import static io.kroxylicious.kubernetes.operator.Labels.standardLabels;
 
 /**
- * A Kube {@code Secret} containing the proxy config YAML.
+ * Generates a Kube {@code Secret} containing the proxy config YAML.
  * We use a {@code Secret} (rather than a {@code ConfigMap})
  * because the config might contain sensitive settings like passwords
  */
 @KubernetesDependent
 public class ProxyConfigSecret
         extends CRUDKubernetesDependentResource<Secret, KafkaProxy> {
 
+    private static final ObjectMapper OBJECT_MAPPER = ConfigParser.createObjectMapper();
```

Comment:
Probably. Let me fill you in on my thinking... I think we want the proxy config to be JSON rather than YAML. That's just a bit more robust against the proxy seeing a partially truncated config file (truncatated JSON is always detectable because of the need for `{}`, `[]` and `""` to balance (well except if your JSON value is just a number, but that's not the case for proxy config)).

But we also need to address the Secret/ConfigMap thing at some point. To do that I'd like for the proxy config to support references between files. So for example you could use a `$ref` (or more likely `$file`) like in JSON Schema to link off to another file on the container file system. That will allow us to mount a config map with an entry-per-route/cluster pattern, and provide a possible route to dynamic reloading by just updating the config map. The job of the operator becomes to mount config maps and secrets, and generate the correct `$file` references between them. The proxy itself and remain relatively dumb and only understand files.

Reply:
> But we also need to address the Secret/ConfigMap thing at some point. 

Absolutely. As someone deploying a Filter CR, I want to be able to keep any secrets (say a Vault token) in a Kubernetes Secret.  That keeps my security department happy.   Is that the kind of use-case you would intend to support?  

The difficulty would be to encode sufficient information into the $ref so it knows what secret/key needs to be resolved.  My initial thought was with an expression `$(secretref:<name>:key)` which could be understood by the operator, secret read and value interpolated.   However, that approach wouldn't benefit from pod projection and it would mean that the operator needs to read a user's secrets. 

Could you expand on your idea?